### DISCUSSION TITLE
Use fake DP ID when using value_redirect mapping?

### DISCUSSION THREAD

**Discussion by GreenSkol** [make-all/tuya-local#1520] [Q&A] [answered]
Hello,

I'm currently writing a configuration file for an [ETOP FCU thermostat](https://www.etopcontrols.com/digital-programmable-heating-and-cooling-fcu-thermostat).

The thermostat can be configured to have its interface in °F or °C and has different DP IDs for °C (temp_set / temp_current) and °F (temp_set_f / temp_current_f).
 
My Home Assistant setup is in °C so my configuration file has the `temperature` & `temperature_current` DPs using the ID of the °C DPs. If I want to use °F, I need to modify the configuration file.

In order to create a generic configuration file, I want to use the `value_redirect` mapping option.
The `temperature` & `temperature_current` entries must still have an ID, but it's unclear which one I should use.
Should I simply put IDs that are not used by my device ?

**Comment by make-all** [comment, marked as answer]
Home Assistant does conversion in the UI, so you don't strictly need to use both dps, you can just use the C dp and mark the unit as C, ignoring the device's setting (which can be made available as "Display unit" if there is a display on the device that you still want to control the unit of).

But if you want the device output to follow the temperature setting to be complete, just choose one of the temperature dps (normally C) as the "temperature" dp, and use a conditional value_redirect to the other dp when temperature_unit is "f".

**Comment by GreenSkol** [comment]
Thanks for the clarification!

So basically there are 2 main ways of configuring, based on the thermostat DPs:

1) The thermostat has DPs that return temperatures in a fixed unit, independently of the thermostat display unit. In this case you  use the `unit` parameter to tell HA how to interpret the data, the default being °C.

2) The thermostat has DPs that can return temperatures either in °C or °F, and has another DP to set/get what unit is being used. In this case you need to configure this DP as `temperature_unit` and HA will automatically control the thermostat with the proper unit.

In my case it's option (1), I've settled on using the DPs in °C and added the DP to set the display unit as a secondary entity.

**Reply by make-all** [reply]
Yes.  Where you have separate C and F dps like your device,has, you can also emulate solution 2 using conditional value_redirect, but it complicates the config, and isn't strictly necessary with HA doing the unit conversion already.  Often on such devices, the F dps will not be sent at all if the unit is set to C.  Some devices may not send the C dps when it is set to F, so the conditional emulation will be required for those to fill the gap.  But often the C dps is primary (usually given a dp id of 2 or 3) and will be sent even when the device is set to F.
